axelor-budget/src/main/java/com/axelor/apps/budget/service/invoice/BudgetInvoiceLineServiceImpl.java

```diff
@@ -153,4 +159,63 @@ public void computeBudgetDistributionSumAmount(InvoiceLine invoiceLine, Invoice
     }
     invoiceLine.setBudgetDistributionSumAmount(budgetDistributionSumAmount);
   }
+
+  @Override
+  public String getBudgetDomain(Invoice invoice, InvoiceLine invoiceLine) {
+    String budget = "self.budgetLevel.parentBudgetLevel.globalBudget";
+    String query =
+        String.format(
+            "self.totalAmountExpected > 0 AND self.statusSelect = %d AND %s.company.id = %d ",
+            BudgetRepository.STATUS_VALIDATED,
+            budget,
+            invoice.getCompany() != null ? invoice.getCompany().getId() : 0);
+
+    if (invoice.getOperationTypeSelect() >= InvoiceRepository.OPERATION_TYPE_CLIENT_SALE) {
+      query =
+          query.concat(
+              String.format(
+                  " AND %s.budgetTypeSelect = %d ",
+                  budget, BudgetLevelRepository.BUDGET_LEVEL_BUDGET_TYPE_SELECT_SALE));
+    } else {
+
+      if (AccountTypeRepository.TYPE_CHARGE.equals(
+          invoiceLine.getAccount().getAccountType().getTechnicalTypeSelect())) {
+        query =
+            query.concat(
+                String.format(
+                    " AND %s.budgetTypeSelect in ("
+                        + BudgetLevelRepository.BUDGET_LEVEL_BUDGET_TYPE_SELECT_PURCHASE
+                        + ","
+                        + BudgetLevelRepository
+                            .BUDGET_LEVEL_BUDGET_TYPE_SELECT_PURCHASE_AND_INVESTMENT
+                        + ")",
+                    budget));
+      } else if (AccountTypeRepository.TYPE_IMMOBILISATION.equals(
+          invoiceLine.getAccount().getAccountType().getTechnicalTypeSelect())) {
+        query =
+            query.concat(
+                String.format(
+                    " AND %s.budgetTypeSelect in ("
+                        + BudgetLevelRepository.BUDGET_LEVEL_BUDGET_TYPE_SELECT_INVESTMENT
+                        + ","
+                        + BudgetLevelRepository
+                            .BUDGET_LEVEL_BUDGET_TYPE_SELECT_PURCHASE_AND_INVESTMENT
+                        + ")",
+                    budget));
+      } else {
+        query = "self.id = 0";
+      }
+    }
```

Comment:
duplicated code with [MoveLineBudgetServiceImpl.java](https://github.com/axelor/axelor-open-suite/pull/11091/files#diff-b5561c56a2cfb2f6234e2009052307856d5f2cd281629d272dc3d2e1daf7cd87) ? Please check if a standard `getBudgetDomain` can be done